### ISSUE TITLE
Can't pay or open channel with "Your current confirmed balance"

### ISSUE THREAD

**Issue by LouBJohnson** [LN-Zap/zap-desktop#3341]
I *now* know I should have played around on testnet before using actual funds. 

Currently can't create a channel or pay. I am not sure atm if this is a bug, a confusing UI issue, or due to my not understanding the landscape I am operating in yet, as it could be one or more of those.

I would like to start out by finding the log file(s) so I can give the proper data for troubleshooting.

Once I get my funds out, I want to start playing around on testnet, with the goal of eventually being able to write and/or contribute to the documentation and perhaps even the UI/UX with "I need ELI5 directions" noobs in mind, as I believe I could make some powerful contributions in that area..

Any help would be greatly appreciated.

**Comment by bolatovumar** [comment]
@LouBJohnson do you get some sort of error when you try to create a channel? Normally the process works like so:
Deposits funds to your wallet --> Open a channel --> Wait for channel opening to confirm --> Make payments.

You might want to join the Zap Slack where you can ask questions and more likely to get help then through Github: https://zaphq.slack.com

**Comment by LouBJohnson** [comment]
> @LouBJohnson do you get some sort of error when you try to create a channel? 

Yes, I've had a few. Im hoping there is a log file I can open that will allow me to say exactly what errors I got and in what order, etc....one of them was "not enough witness outputs to create funding transaction"

Normally the process works like so:
> Deposits funds to your wallet --> Open a channel --> Wait for channel opening to confirm 
I might have screwed the pooch on the waiting for it to confirm part. I waited a while, and closed it (cooperatively and sucessfully, afaict) while it was still pending. Other UI weirdness ensued, for ex my balance temporarily showed as more than the initial btc funduing amount....closed and re-opened zap and that went down to a more appropriate amount, but still getting errors, that one and one other that I was too flustered and impatient at the time to document.
 
> You might want to join the Zap Slack where you can ask questions and more likely to get help then through Github: https://zaphq.slack.com

I posted in the slack (different screen name, gary briane tuttle) with a "hello world" and a request for the location of any log files. I think I posted there, anyway. The list of known unknowns and potential unknown unknowns in this whole process has got me past brain shutdown mode....in any case, I do not see anything that could be a reply to that in the slack.

with that in mind...I would still like to get my information correct and learn some in the process by going over log files, if they exist, so...

**_Is there a log file created by the Windows desktop app that I could use to expedite the communication of what steps were taken, and the current state of the pertinent "moving parts", if so, where is it?_**

**Comment by bolatovumar** [comment]
@LouBJohnson see this comment for a hint on where you might find your log files on Windows:
https://github.com/LN-Zap/zap-desktop/issues/3334#issuecomment-579980867

> "not enough witness outputs to create funding transaction"

I have seen this error before not in Zap but other LN apps. This error comes directly from LND. It means that LND can't create a funding transaction as there are not enough confirmed Segwit outputs. In your case this may mean that you either need to wait for your transaction to your Zap wallet to confirm or if it's already confirmed, try opening a smaller channel.

**Comment by LouBJohnson** [comment]
> @LouBJohnson see this comment for a hint on where you might find your log files on Windows:
> [#3334 (comment)](https://github.com/LN-Zap/zap-desktop/issues/3334#issuecomment-579980867)
> 
> > "not enough witness outputs to create funding transaction"
> 
> I have seen this error before not in Zap but other LN apps. This error comes directly from LND. It means that LND can't create a funding transaction as there are not enough confirmed Segwit outputs. In your case this may mean that you either need to wait for your transaction to your Zap wallet to confirm or if it's already confirmed, try opening a smaller channel.

Thanks!

 That does seem to be part of the issues. I do believe I am experiencing a multivariate issue, which,  tbh, I exacerbated in my haste combined with a lack of awareness of the wild fluctuations in my cognitive processing power.

Im currently working to untangle the data and will report back.

thanks again

**Comment by bolatovumar** [comment]
@LouBJohnson try sending all of your funds to some other wallet you control, create a new wallet in Zap and send the funds to it again and try to open some channels once the transaction confirms. Also, do not enable autopilot if you want to be manually opening channels.

**Comment by LouBJohnson** [comment]
> @LouBJohnson try sending all of your funds to some other wallet you control, create a new wallet in Zap and send the funds to it again and try to open some channels once the transaction confirms. Also, do not enable autopilot if you want to be manually opening channels.

Thank you.

So, just so I get it straight, is this what I should do?

1) Create a new wallet in Zap

2) Get LN payment address in "Receive" from **_New_** zap wallet (**So, **no** channel opening needed?**)

3) Send **_entire_** amount currently listed as "Your current confirmed balance" to address I got from step 2 (for clarity, this amount in **_Old_** wallet is listed as"_**1,106,879 satoshis (onchain), 0 satoshis (in channels)**_"...or **_do I need to account for some Tx Fees at this step?_**)

4) Open a new channel in **_New_** wallet, funding the channel with the maximum amount I can without getting the "not enough witnesses" error, which will be total amount, minus BTC Tx fees

5) Send whatever amount I am going to send to the Lightning enabled merchant, and/or a custodial Lightning wallet (for example like Phoenix)
and at that point,
6) I have now "cleaned out" my first and second Zap wallets, and can start playing around on testnet without leaving actual funds anywhere

**Comment by bolatovumar** [comment]
> 1. Create a new wallet in Zap

Yes

> 2. Get LN payment address in "Receive" from **_New_** zap wallet (**So, **no** channel opening needed?**)

No, you need an on-chain receive address. You can get it by clicking in the QR code looking button in the top left corner.

![Screen Shot 2020-03-02 at 7 27 49 PM](https://user-images.githubusercontent.com/1934678/75740254-070b2780-5cbc-11ea-9569-02ccf949d4bf.png)

You are right, you are not opening any channels yet at this point, just transferring funds to your on-chain wallet.

> 3. Send **_entire_** amount currently listed as "Your current confirmed balance" to address I got from step 2 (for clarity, this amount in **_Old_** wallet is listed as"_**1,106,879 satoshis (onchain), 0 satoshis (in channels)**_"...or **_do I need to account for some Tx Fees at this step?_**)

When transferring your funds to your new wallet you can click on "Send all of your funds" toggle to send all of your funds.

![Screen Shot 2020-03-02 at 7 30 47 PM](https://user-images.githubusercontent.com/1934678/75740410-65380a80-5cbc-11ea-8abf-8667c37fb5c9.png)

> 4. Open a new channel in **_New_** wallet, funding the channel with the maximum amount I can without getting the "not enough witnesses" error, which will be total amount, minus BTC Tx fees

Yup

> 5. Send whatever amount I am going to send to the Lightning enabled merchant, and/or a custodial Lightning wallet (for example like Phoenix)

If at this point you have your channel opened and it can find a route to your payment destination node then you should be able to make payments to it.

>    and at that point,
> 6. I have now "cleaned out" my first and second Zap wallets, and can start playing around on testnet without leaving actual funds anywhere

At this point all of your funds from your first wallet should be in the new wallet you created in step 1 with new channel(s) open and you should be able to make lightning payments.

**Comment by LouBJohnson** [comment]
> > 1. Create a new wallet in Zap
> 
> Yes
> 
> > 1. Get LN payment address in "Receive" from **_New_** zap wallet (**So, **no** channel opening needed?**)
> 
> No, you need an on-chain receive address. You can get it by clicking in the QR code looking button in the top left corner.
> 
> ![Screen Shot 2020-03-02 at 7 27 49 PM](https://user-images.githubusercontent.com/1934678/75740254-070b2780-5cbc-11ea-9569-02ccf949d4bf.png)
> 
> You are right, you are not opening any channels yet at this point, just transferring funds to your on-chain wallet.
> 
> > 1. Send **_entire_** amount currently listed as "Your current confirmed balance" to address I got from step 2 (for clarity, this amount in **_Old_** wallet is listed as"_**1,106,879 satoshis (onchain), 0 satoshis (in channels)**_"...or **_do I need to account for some Tx Fees at this step?_**)
> 
> When transferring your funds to your new wallet you can click on "Send all of your funds" toggle to send all of your funds.
> 
> ![Screen Shot 2020-03-02 at 7 30 47 PM](https://user-images.githubusercontent.com/1934678/75740410-65380a80-5cbc-11ea-8abf-8667c37fb5c9.png)
> 
> > 1. Open a new channel in **_New_** wallet, funding the channel with the maximum amount I can without getting the "not enough witnesses" error, which will be total amount, minus BTC Tx fees
> 
> Yup
> 
> > 1. Send whatever amount I am going to send to the Lightning enabled merchant, and/or a custodial Lightning wallet (for example like Phoenix)
> 
> If at this point you have your channel opened and it can find a route to your payment destination node then you should be able to make payments to it.
> 
> > and at that point,
> > 6. I have now "cleaned out" my first and second Zap wallets, and can start playing around on testnet without leaving actual funds anywhere
> 
> At this point all of your funds from your first wallet should be in the new wallet you created in step 1 with new channel(s) open and you should be able to make lightning payments.

Thanks so much for your replies and attention. I will re-read everything again before I give this a shot tomorrow, but I do think I am getting a LOT of understanding of the overall flow and the UI as a result of this dialog. (_think_ as vs _know_ I am getting the understanding, hence the waiting until tomorrow)

Good stuff. Thanks again.

**Comment by LouBJohnson** [comment]
Ok, so,

Here is an update, with the current place I am stuck:

I went through the process with no problems, until trying to open a channel through fold's node. I kept getting a funding error. I lowered the amount I was trying to fund the channel with way past anything reasonable, but no luck.

I then opened one through Zap's node with an amount that had previously failed through Fold's, and it worked.

Next, Error when I tried to buy a gift card by paying Fold (tried it twice) Error is: "Unable to route payment to destination" (Complete error text is below)

My initial reading on it seems to indicate this is some kind of channel capacity (liquidity?) issue...Is it?

Also, is a specific cause somewhere indicated in the full error message? If so, where?

I will try to pay again a bit later with a new payment request 

Thanks!

Error message text:

Unable to route payment to destination: TemporaryChannelFailure(update=(*lnwire.ChannelUpdate)(0xc002446210)({
 Signature: (lnwire.Sig) (len=64 cap=64) {
  00000000  c8 63 17 48 ce 9a 13 5b  89 14 1f e2 12 82 e4 c5  |.c.H...[........|
  00000010  92 21 8d b0 51 1d 1c 0f  76 ad af e8 b0 9e bf 3f  |.!..Q...v......?|
  00000020  21 ac bb 15 04 48 a0 0c  b7 27 54 d9 ba ae c9 a0  |!....H...'T.....|
  00000030  89 63 af 1a 6c ab e2 f3  05 18 c4 99 70 24 8d 3e  |.c..l.......p$.>|
 },
 ChainHash: (chainhash.Hash) (len=32 cap=32) 000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f,
 ShortChannelID: (lnwire.ShortChannelID) 607147:2312:1,
 Timestamp: (uint32) 1583321002,
 MessageFlags: (lnwire.ChanUpdateMsgFlags) 00000001,
 ChannelFlags: (lnwire.ChanUpdateChanFlags) 00000001,
 TimeLockDelta: (uint16) 40,
 HtlcMinimumMsat: (lnwire.MilliSatoshi) 1000 mSAT,
 BaseFee: (uint32) 1000,
 FeeRate: (uint32) 1,
 HtlcMaximumMsat: (lnwire.MilliSatoshi) 2475000000 mSAT,
 ExtraOpaqueData: ([]uint8) <nil>
})
)

**Comment by LouBJohnson** [comment]
Now getting "Unable to route payment to destination: UnknownNextPeer"

**Comment by bolatovumar** [comment]
@LouBJohnson "Unable to route payment to destination:" literally means that LND can't find a route to whatever node it is you are trying to pay. This means you need to open a channel or two to some well-connected nodes. You can find well-connected nodes to open a channel with here: https://1ml.com

**Comment by LouBJohnson** [comment]
> @LouBJohnson "Unable to route payment to destination:" literally means that LND can't find a route to whatever node it is you are trying to pay. This means you need to open a channel or two to some well-connected nodes. You can find well-connected nodes to open a channel with here: https://1ml.com

Thanks.

 A couple questions, to empower my research:

_So, this means that for some reason, the _**Zap node**_ is unable to find a route (i.e., that the LN Node is the cause of the issue, or the sticking point)?_

_Are there resources you could point me to that explain the overall concepts in use in routing payments through the lightning network?_ 

I have started this research, but have found more BS bcash propaganda about 'the routing issue will not ever be solved' and 'big banks have bought bitcoin by controlling the LN' than actual conceptual explanations of how LN routing works. (Learning about this is exciting !)

Thanks again!

I think I should probably close this issue (on github I mean) once/if I hear back from you, right?  My initial problem is solved, for sure.

**Comment by bolatovumar** [comment]
@LouBJohnson you can close this issue once you successfully achieve your goal.

> So, this means that for some reason, the Zap node is unable to find a route (i.e., that the LN Node is the cause of the issue, or the sticking point)?

Yeah, most likely this is the case. I'm guessing the node you opened a channel with is just not connected to any node which has a path towards your destination node that you want to make a payment to.

> Are there resources you could point me to that explain the overall concepts in use in routing payments through the lightning network?

Take a look at this: https://github.com/lnbook/lnbook. It's very much still work in progress but will probably answer some of your questions.

Take a look at videos on this channel: https://www.youtube.com/user/renepickhardt/videos

Take a look at this: https://github.com/bitcoinbook/bitcoinbook/blob/2db79a8001500530e04114643a80e2cf73d8fa19/ch12.asciidoc#ln_payment_process

Conceptually routing is pretty simple. As long as there is a path between your node and the node you are trying to make a payment to and there is enough channel capacity along this path you should be able to make a payment.

**Comment by LouBJohnson** [comment]
all set on this one, thanks.

I discovered some other stuff in the proces and hopefully will be able to carve out time to see if there are already open issues on themt and/or open issues if there are not.

thanks again

**Comment by LouBJohnson** [comment]
JUST seeing this now, Thank you!
